src/materialized/src/http/root.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! HTTP endpoints for the homepage and static files.
+
+use std::future::Future;
+
+use askama::Template;
+use futures::future;
+use hyper::{Body, Method, Request, Response, StatusCode};
+use include_dir::{include_dir, Dir};
+
+use crate::http::{util, Server};
+
+#[derive(Template)]
+#[template(path = "http/templates/home.html")]
+struct HomeTemplate<'a> {
+    version: &'a str,
+    build_time: &'a str,
+    build_sha: &'static str,
+}
+
+const STATIC_DIR: Dir = include_dir!("src/http/static");
+
+impl Server {
+    pub fn handle_home(
```

Comment:
why can't these be async functions?

Reply:
I guess the lifetime checker doesn't like async functions that take non-static references (even if their use doesn't actually cross an `.await`)

Reply:
Yeah, exactly; the lifetime of the future returned by an `async fn` always captures all of the input lifetimes, which we explicitly do not want to do here. This is called the "initialization pattern" in the RFC (https://rust-lang.github.io/rfcs/2394-async_await.html#initialization-pattern).